src/commands.ts

```diff
@@ -91,6 +92,139 @@ export async function updateDependencies(ctx: WorkspaceContext) {
+        }
+        initialValue = `swift-${projectType}-${i}`;
+    }
+    const projectName = await vscode.window.showInputBox({
+        value: initialValue,
+        prompt: "Enter a name for your new swift project",
+        validateInput(value) {
+            // Ensure there are no name collisions
+            if (existingNames.includes(value)) {
+                return "A file/folder with this name already exists";
+            }
+            return undefined;
+        },
+    });
+    if (!projectName) {
+        return undefined;
+    }
+
+    // Create the folder that will store the new project
+    const projectUri = vscode.Uri.joinPath(selectedFolder[0], projectName);
+    await fs.mkdir(projectUri.fsPath);
+
+    // Use swift package manager to initialize the swift project
+    await withDelayedProgress(
+        {
+            location: vscode.ProgressLocation.Notification,
+            title: `Creating swift project ${projectName}`,
+        },
+        async () => {
+            await execSwift(
+                ["package", "init", "--type", projectType, "--name", projectName],
+                ctx.toolchain,
+                {
+                    cwd: projectUri.fsPath,
+                }
+            );
+        },
+        1000
+    );
+
+    // Prompt the user whether or not they want to open the newly created project
+    const isWorkspaceOpened = !!vscode.workspace.workspaceFolders;
+    const config = vscode.workspace.getConfiguration("swift");
+    const openAfterCreate = config.get<string>("openAfterCreateProject");
+
+    let action: "open" | "openNewWindow" | "addToWorkspace" | undefined;
+    if (openAfterCreate === "always") {
+        action = "open";
+    } else if (openAfterCreate === "alwaysNewWindow") {
+        action = "openNewWindow";
+    } else if (openAfterCreate === "whenNoFolderOpen" && !isWorkspaceOpened) {
+        action = "open";
+    }
+
+    if (action === undefined) {
+        let message = `Would you like to open ${projectName}?`;
+        const open = "Open";
+        const openNewWindow = "Open in New Window";
+        const choices = [open, openNewWindow];
+
+        const addToWorkspace = "Add to Workspace";
+        if (isWorkspaceOpened) {
+            message = `Would you like to open ${projectName}, or add it to the current workspace?`;
+            choices.push(addToWorkspace);
+        }
+
+        const result = await vscode.window.showInformationMessage(
+            message,
+            { modal: true, detail: "The default action can be configured in settings" },
+            ...choices
+        );
+        if (result === open) {
+            action = "open";
+        } else if (result === openNewWindow) {
+            action = "openNewWindow";
+        } else if (result === addToWorkspace) {
+            action = "addToWorkspace";
+        }
+    }
+
+    if (action === "open") {
+        vscode.commands.executeCommand("vscode.openFolder", projectUri, { forceReuseWindow: true });
```

Comment:
suggestion: can we await or .catch to not leave a floating promise?

Reply:
Good catch!